File: exercise/php/day24/src/Delivery/Domain/Core/RegisteredRoutes.php
<?php

namespace Delivery\Domain\Core;

class RegisteredRoutes
{
    private array $routes = [];

    public function dispatch(Event $event): void
    {
        $eventType = get_class($event);
        if (isset($this->routes[$eventType])) {
            $this->routes[$eventType]($event);
        }
    }

    public function register(string $eventType, callable $apply): void
    {
        $this->routes[$eventType] = $apply;
    }
}

File: exercise/C#/day24/src/Delivery/Domain/Core/RegisteredRoutes.cs
using LanguageExt;

namespace Delivery.Domain.Core
{
    internal class RegisteredRoutes
    {
        private Map<string, Action<IEvent>> _routes = new();

        public void Dispatch<TEvent>(TEvent @event) where TEvent : IEvent
            => _routes[@event.GetType().ToString()](@event);

        public void Register(string eventType, Action<IEvent> apply)
            => _routes = _routes.Add(eventType, apply);
    }
}